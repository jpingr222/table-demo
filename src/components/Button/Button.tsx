import styled, { ThemeProvider } from "styled-components"

const StyledButton = styled.button`
  font-size: ${({theme}) => theme.fontSize};
  line-height: ${({theme}) => theme.lineHeight};
  color: ${({theme}) => theme.color};
  display: inline-block;
  text-align: center;
  cursor: pointer;
  box-sizing: border-box;
  background-color: ${({theme}) => theme.backgroundColor};
  margin: 0;
  padding: ${({theme}) => theme.padding};
  border: ${({theme}) => theme.border};
  border-radius: 0.25rem;

  &:hover {
    color: ${({theme}) => theme.hoverColor};
    background-color: ${({theme}) => theme.hoverBackgroundColor};
  }
`;

const buttonTheme = {
  normal: {
    fontSize: '1rem',
    lineHeight: '1.5',
    color: '#2d3339',
    backgroundColor: '#fff',
    padding: '0.25rem 0.75rem',
    border: '1px solid #2d3339',
    hoverColor: '#fff',
    hoverBackgroundColor: '#2d3339'
  },
  close: {
    fontSize: '2.5rem',
    lineHeight: '1.5rem',
    color: 'rgba(0, 0, 0, 0.4)',
    backgroundColor: 'transparent',
    padding: '0.25rem',
    border: '0',
    hoverColor: 'rgba(0, 0, 0, 0.6)',
    hoverBackgroundColor: 'transparent'
  }
};

export interface ButtonProps {
  theme?: 'normal' | 'close';
  children?: React.ReactNode;
  onClick?: () => void;
}

export default function Button({
  theme = 'normal',
  children,
  onClick
}: ButtonProps) {
  return (
    <ThemeProvider theme={buttonTheme[theme]}>
      <StyledButton onClick={onClick}>{children}</StyledButton>
    </ThemeProvider>
  );
}
