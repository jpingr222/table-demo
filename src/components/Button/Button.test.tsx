import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Button from "./Button";

describe('Button test', () => {
  it('should render an empty button correctly', () => {
    render(<Button></Button>);

    const button: HTMLButtonElement = screen.getByRole('button');

    expect(button).toMatchSnapshot();
  });

  it('should render a button with correct style', () => {
    render(<Button theme="close"></Button>);

    const button: HTMLButtonElement = screen.getByRole('button');

    expect(button).toMatchSnapshot();
  });

  it('should render a button with text', () => {
    render(<Button>Button</Button>);

    const button: HTMLButtonElement = screen.getByRole('button');

    expect(button).toHaveTextContent('Button');
  });

  it('should render a button with text', () => {
    const onClick = jest.fn();
    render(<Button onClick={onClick}>Button</Button>);

    const button: HTMLButtonElement = screen.getByRole('button');
    userEvent.click(button);

    expect(onClick).toBeCalledTimes(1);
  });

});
