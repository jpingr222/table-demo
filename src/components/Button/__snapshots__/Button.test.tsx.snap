// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test should render a button with correct style 1`] = `
<button
  class="sc-bcXHqe iZcHbs"
/>
`;

exports[`Button test should render an empty button correctly 1`] = `
<button
  class="sc-bcXHqe fzkNmB"
/>
`;
